content-store/src/main/java/nl/inl/blacklab/contentstore/ContentStoreFixedBlockWriter.java

```diff
@@ -211,23 +211,30 @@ protected void mapToc(boolean writeable) throws IOException {
         } // leave 1M room at the end
         tocFileChannel = tocRaf.getChannel();
         tocFileBuffer = tocFileChannel.map(writeable ? MapMode.READ_WRITE : MapMode.READ_ONLY, 0, fl);
+        double flsize = (double) fl / (1024 * 1024);
+        logger.debug("Memory mapping toc file, file size={}mb, toc size={}", flsize, toc.size() );
     }
 
+
     private void writeToc() {
         try {
             mapToc(true);
             tocFileBuffer.putInt(toc.size());
             try {
+                long tocSizeInBytes = 0;
                 for (TocEntry e : toc.values()) {
+                    tocSizeInBytes += e.sizeBytes();
                     if (tocFileBuffer.remaining() < e.sizeBytes()) {
                         // Close and re-open with extra writing room
                         int p = tocFileBuffer.position();
+                        logger.debug("Closing and reopening with extra room, position={}", p);
                         closeMappedToc();
                         mapToc(true);
                         ((Buffer)tocFileBuffer).position(p);
                     }
                     e.serialize(tocFileBuffer);
                 }
+                logger.debug("Wrote a toc of size={} mb", tocSizeInBytes / (1024 * 1024));
```

Comment:
You may want to cast this to double too.

Reply:
I just need rough numbers for now. This will work